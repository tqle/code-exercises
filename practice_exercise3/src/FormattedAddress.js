import React from "react";
import {
  validateState,
  validateStreet,
  validateSuite,
  validateCity,
  validateZipCode
} from "./utils/addressUtil";

/**
 * Formatted Address
 * Custom format address content in the HTML Address element
 */
class FormattedAddress extends React.Component {
  render() {
    // Destructuring objects from props.
    const { street, suite, state, city, zipcode } = this.props;
    //  User address utility to validate or format any address properties
    // for undefined or nulls.
    let validStreet, validSuite, validState, validCity, validZipCode;
    // Check for empty, undefined, null cases
    validState = validateState(state);
    validStreet = validateStreet(street);
    validSuite = validateSuite(suite);
    validCity = validateCity(city);
    validZipCode = validateZipCode(zipcode);

    // Build address in the designated format

    return (
      <address className="normal-style">
        {validStreet} {validSuite} <br />
        {validCity}, {validState}, {validZipCode}
      </address>
    );
  }
}

export default FormattedAddress;
