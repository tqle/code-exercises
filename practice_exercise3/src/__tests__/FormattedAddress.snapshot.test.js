import React from "react";
import { create } from "react-test-renderer";
import FormattedAddress from "../FormattedAddress";

/**
 * Test Email Link snapshot
 */
test("<FormattedAddress /> renders correctly", () => {
  const street = "Kulas Light";
  const suite = "Apt. 556";
  const city = "Gwenborough";
  const state = "FL";
  const zipCode = "92998-3874";

  const test = create(
    <FormattedAddress
      street={street}
      suite={suite}
      city={city}
      state={state}
      zipCode={zipCode}
    />
  );
  expect(test.toJSON()).toMatchSnapshot();
});

test("<FormattedAddress /> renders correctly with no suite", () => {
  const street = "Kulas Light";
  const suite = "";
  const city = "Gwenborough";
  const state = "FL";
  const zipCode = "92998-3874";

  const test = create(
    <FormattedAddress
      street={street}
      suite={suite}
      city={city}
      state={state}
      zipCode={zipCode}
    />
  );
  expect(test.toJSON()).toMatchSnapshot();
});

test("<FormattedAddress /> renders an empty address", () => {
  const street = "";
  const suite = "";
  const city = "";
  const state = "";
  const zipCode = "";
  const test = create(
    <FormattedAddress
      street={street}
      suite={suite}
      city={city}
      state={state}
      zipCode={zipCode}
    />
  ).toJSON();
  expect(test).toMatchSnapshot();
});
