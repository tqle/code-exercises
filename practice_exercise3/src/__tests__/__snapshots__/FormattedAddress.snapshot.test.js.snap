// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedAddress /> renders an empty address 1`] = `
<address
  className="normal-style"
>
  
   
  
   
  <br />
  
  , 
  FL
  , 
  
</address>
`;

exports[`<FormattedAddress /> renders correctly 1`] = `
<address
  className="normal-style"
>
  Kulas Light
   
  Apt. 556
   
  <br />
  Gwenborough
  , 
  FL
  , 
  
</address>
`;

exports[`<FormattedAddress /> renders correctly with no suite 1`] = `
<address
  className="normal-style"
>
  Kulas Light
   
  
   
  <br />
  Gwenborough
  , 
  FL
  , 
  
</address>
`;
